### PR TITLE
Improve specs and guards in Code module

### PR DESCRIPTION
**Pull request by eksperimental**

### CODE REVIEW

**Comment by whatyouhide** [comment]
For who's gonna merge this: maybe a better commit message would be "Improve specs and guards in the Code module" :).

File: lib/elixir/lib/code.ex [hunk 1]
```diff
@@ -509,6 +509,8 @@ defmodule Code do
       {:error, :nofile}
 
   """
+  @spec ensure_loaded(module) ::
+      {:module, module} | {:error, :embedded | :badfile | :nofile | :on_load_failure}
```
Comment (whatyouhide) [inline lib/elixir/lib/code.ex]: Is it common to indent this 4 spaces?

**Comment by eksperimental** [comment]
@whatyouhide I've been doing commit message like this: 
- Module: comment
- Module.function/arity: comment

I know this case in particular could be confusing, I could change it to:
- Code module: improve specs and guards

if you agree with it

File: lib/elixir/lib/code.ex (hunk 1 above)
Comment (eksperimental) [inline lib/elixir/lib/code.ex]: fixed

File: lib/elixir/lib/code.ex [hunk 2]
```diff
@@ -509,6 +509,8 @@ defmodule Code do
       {:error, :nofile}
 
   """
+  @spec ensure_loaded(module) ::
+    {:module, module} | {:error, :embedded | :badfile | :nofile | :on_load_failure}
```
Comment (josevalim) [inline lib/elixir/lib/code.ex]: The indentation should align with the letter `e`.

File: lib/elixir/lib/code.ex [hunk 3]
```diff
@@ -543,6 +545,8 @@ defmodule Code do
   Check `ensure_loaded/1` for more information on module loading
   and when to use `ensure_loaded/1` or `ensure_compiled/1`.
   """
+  @spec ensure_compiled(module) ::
+    {:module, module} | {:error, :embedded | :badfile | :nofile | :on_load_failure}
```
Comment (josevalim) [inline lib/elixir/lib/code.ex]: Same here.

**Comment by whatyouhide** [comment]
@eksperimental not really, I am 👎 on this kind of commits. The reason is that we have no standard in the Elixir code base, so only your commits will look like this, which I think will make things slightly confusing. The only standard I think we have in Elixir is to commit with a sentence that reads nicely, which is at least what I always try to do, and I think it's good because it's not structured and kinda natural. Wdyt?

**Comment by eksperimental** [comment]
@whatyouhide I can do that. no worries. 
as for the reasons behind, for me it's easier much easier to scan the first word of a commit message to know what the commit is about, rather than having to get to the end of the message

File: lib/elixir/lib/code.ex (hunk 3 above)
Comment (eksperimental) [inline lib/elixir/lib/code.ex]: fixed.

**Comment by whatyouhide** [comment]
@eksperimental yes, but as I said we have no convention to do so so it's not very useful anyhow. Also, it doesn't scale really well (e.g., changes in multiple modules and so on). :)

**Comment by josevalim** [comment]
:heart: :green_heart: :blue_heart: :yellow_heart: :purple_heart:

**Pull request merged**